packages/server/graphql/mutations/helpers/inviteToTeamHelper.ts

```diff
@@ -78,11 +37,30 @@ const inviteToTeamHelper = async (
   const operationId = dataLoader.share()
   const subOptions = {mutatorId, operationId}
 
-  const trustScore = await getInviteTrustScore(viewerId, teamId, invitees, dataLoader)
-  if (trustScore < 0.15)
-    return {error: {message: 'Cannot invite by email. Try using invite link'}, invitees: []}
+  const untrustedDomains = ['tempmail.cn', 'qq.com']
+  const filteredInvitees = invitees.filter(
+    (invitee) => !untrustedDomains.includes(getDomainFromEmail(invitee).toLowerCase())
+  )
+  const mailManager = new MailManagerMailgun()
+  const validateEmailRes = await Promise.all(
+    invitees.map((invitee) => mailManager.validateEmail(invitee))
+  )
+  const validInvitees = filteredInvitees.filter((invitee, index) => {
+    const validationResult = validateEmailRes[index]
+    if (!validationResult) return false
+    if (validationResult === 'undeliverable' || validationResult === 'do_not_send') {
+      const error = new Error(`Email validation error: ${validationResult}`)
+      sendToSentry(error, {tags: {invitee}})
+      return false
+    }
+    return true
+  })
```

Comment:
+1 a slightly cleaner way than iterating through the list twice and matching by index would be to do sth. like this (pseudo code):
```ts
const validInvitess = (await Promise.all(
  invitess.map(async (invitee) => {
    const validationResult = await mailManager.validateEmail(invitee))
    if (!validationResult || validationResult === 'undeliverable'...) {
     ..
     return error
    }
    return invitee
  })
)).filter(isValid)
```